bookmarks/vanilla/README.md

```diff
@@ -14,10 +15,95 @@ npm run test
+
+Once its completed the auth proccess it will redirect to the redirect url, and from there it will use Bookmark static methods to CRUD the bookmarks in [Bookmarks component](https://github.com/solid-contrib/data-modules/blob/main/bookmarks/vanilla/demo/src/components/Bookmarks/Bookmarks.tsx)
+
+
+These are the methods implemented to deal with bookmarks
+They will return a promise containing the JSON representation of the bookmark
+
+```typescript
+
+import { Session } from "@inrupt/solid-client-authn-browser";
+
+const result: IBookmark[] = await Bookmarks.getAll(session: Session);
+const result: IBookmark  = await Bookmarks.get("<pk>", session: Session);
+const result: boolean  = await Bookmarks.remove("<pk>", session: Session);
+const result: IBookmark  = await Bookmarks.create(payload: ICreateBookmark, session: Session);
+const result: IBookmark  = await Bookmarks.update("<pk>", payload: IUpdateBookmark, session: Session);
+```
+
+- pk indicates the primary key of the bookmark, in this case its the url of the bookmark
+- payload is an object containing the fields of the bookmark
+- you need to pass the authenticated session to the methods, this session is obtained from the [auth.ts](https://github.com/solid-contrib/data-modules/blob/main/bookmarks/vanilla/demo/src/utils/auth.ts)
+
+
+## Types
+
+```typescript
+type ICreateBookmark = {
+    title: string
+    topic?: string
+    link: string
+    creator?: string,
+}
+
+type IUpdateBookmark = {
+    title: string
+    topic?: string
+    link: string
+    creator?: string,
+}
+
+type IBookmark = ICreateBookmark & {
+    url: string
+    created?: string
+    updated?: string
+}
+```
+
+
+## valid payload
+the [`Bookmark.getAll`](https://github.com/solid-contrib/data-modules/blob/422cabb91085916e71c5610235f43fc483493d72/bookmarks/vanilla/src/modules/Bookmark.ts#L72) takes an authenticated session to use for fetching the data.
+
+the [`Bookmark.get`](https://github.com/solid-contrib/data-modules/blob/422cabb91085916e71c5610235f43fc483493d72/bookmarks/vanilla/src/modules/Bookmark.ts#L94) and [`Bookmark.delete`](https://github.com/solid-contrib/data-modules/blob/422cabb91085916e71c5610235f43fc483493d72/bookmarks/vanilla/src/modules/Bookmark.ts#L108) methods take the url of the bookmark as the primary key and an authenticated session.
+
+the [`Bookmark.create`](https://github.com/solid-contrib/data-modules/blob/422cabb91085916e71c5610235f43fc483493d72/bookmarks/vanilla/src/modules/Bookmark.ts#L135) takes an object with fields: `label`, `topic` and `link`, `creator`, `created` and `updated` as the payload with an authenticated
+
+the [`Bookmark.update`](https://github.com/solid-contrib/data-modules/blob/422cabb91085916e71c5610235f43fc483493d72/bookmarks/vanilla/src/modules/Bookmark.ts#L169) takes the `url` of the bookmark as primary key, then as a payload, it takes an object with fields: `label`, `topic` and `link`, `creator`, `created` and `updated` with an authenticated session.
```

Comment:
Please add a description of what these methods return. And for example in case of `update` you'd also have to mention that it will return `undefined` in case the object does not exist.

Reply:
<img width="1091" alt="Screenshot 2023-11-29 at 16 52 19" src="https://github.com/solid-contrib/data-modules/assets/53468377/48ed2e1d-2cb2-418c-9835-dc1b28b10f63">

Reply:
Return types in the readme are also mentioned in

<img width="1093" alt="Screenshot 2023-11-29 at 16 56 30" src="https://github.com/solid-contrib/data-modules/assets/53468377/9c4dbdee-c586-4634-b02c-179e9ff94ff7">

Reply:
![image](https://github.com/solid-contrib/data-modules/assets/4597931/788feb81-b8ec-4731-af98-1bc947da303f)
Please check this sample of what I mean by documenting the return value.

Reply:
Let's leave this as a follow-up, so Reza can merge his PR -> https://github.com/solid-contrib/data-modules/issues/37